datasets/tsac/tsac.py

```diff
@@ -0,0 +1,114 @@
+@inproceedings{medhaffar-etal-2017-sentiment,
+    title = "Sentiment Analysis of {T}unisian Dialects: Linguistic Ressources and Experiments",
+    author = "Medhaffar, Salima  and
+      Bougares, Fethi  and
+      Est{`e}ve, Yannick  and
+      Hadrich-Belguith, Lamia",
+    booktitle = "Proceedings of the Third {A}rabic Natural Language Processing Workshop",
+    month = apr,
+    year = "2017",
+    address = "Valencia, Spain",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/W17-1307",
+    doi = "10.18653/v1/W17-1307",
+    pages = "55--61",
+    abstract = "Dialectal Arabic (DA) is significantly different from the Arabic language taught in schools and used in written communication and formal speech (broadcast news, religion, politics, etc.). There are many existing researches in the field of Arabic language Sentiment Analysis (SA); however, they are generally restricted to Modern Standard Arabic (MSA) or some dialects of economic or political interest. In this paper we are interested in the SA of the Tunisian Dialect. We utilize Machine Learning techniques to determine the polarity of comments written in Tunisian Dialect. First, we evaluate the SA systems performances with models trained using freely available MSA and Multi-dialectal data sets. We then collect and annotate a Tunisian Dialect corpus of 17.000 comments from Facebook. This corpus allows us a significant accuracy improvement compared to the best model trained on other Arabic dialects or MSA data. We believe that this first freely available corpus will be valuable to researchers working in the field of Tunisian Sentiment Analysis and similar areas.",
+}
+"""
+
+_TRAIN_POS_URL = "https://raw.githubusercontent.com/fbougares/TSAC/master/train_pos.txt"
+_TRAIN_NEG_URL = "https://raw.githubusercontent.com/fbougares/TSAC/master/train_neg.txt"
+_TEST_POS_URL = "https://raw.githubusercontent.com/fbougares/TSAC/master/test_pos.txt"
+_TEST_NEG_URL = "https://raw.githubusercontent.com/fbougares/TSAC/master/test_neg.txt"
+
+
+class TSAC(datasets.GeneratorBasedBuilder):
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "sentence": datasets.Value("string"),
+                    "target": datasets.ClassLabel(names=["1", "-1"]),
+                },
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE_URL,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        train_pos_path = dl_manager.download_and_extract(_TRAIN_POS_URL)
+        train_neg_path = dl_manager.download_and_extract(_TRAIN_NEG_URL)
+        test_pos_path = dl_manager.download_and_extract(_TEST_POS_URL)
+        test_neg_path = dl_manager.download_and_extract(_TEST_NEG_URL)
```

Comment:
just for info you can use a dict like 
```python
{
    "train_pos": _TRAIN_POS_URL,
    "train_neg": _TRAIN_NEG_URL,
    "test_pos": _TEST_POS_URL,
    "test_neg": _TEST_NEG_URL
}
```
and pass it to the `dl_manager.download_and_extract` function, which returns the dictionary of paths as expected.

This is useful when you have multiple downloads since they can be run in parallel

Reply:
Will keep it this way since the dataset is small and take care of this in future for large datasets. merging.

Reply:
sounds good to me